### PR TITLE
4 signup

### PR DESCRIPTION
**Pull request by AhmedMughessib**

### CODE REVIEW

**Review by abeerAyad — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]
its still need more fix and more check also needs validation but for now its ok

File: src/controllers/user/signup.ts [hunk 1]
```diff
@@ -1,22 +1,26 @@
 import type { Request, Response } from 'express';
+import { hash } from 'bcrypt';
 import User from '../../database/user';
+// import hashPassword from '../../helpers/hashPromises';
 
 const signupController = async (req: Request, res: Response): Promise<void> => {
```
Comment (AdhamHaisami) [inline src/controllers/user/signup.ts]: req type must not be Request, it should be the interface we built

File: src/controllers/user/signup.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-      hieght,
+      password: hashedPassword,
+      weight,
+      height,
       gender,
-      goalWieght,
+      goalWeight,
       role,
     });
+    console.log(hashedPassword);
```
Comment (AdhamHaisami) [inline src/controllers/user/signup.ts]: delete the console log

File: src/controllers/user/signup.ts [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
     res.status(201).json({ massage: 'user created successfully' });
   } catch (error) {
-    console.log(error);
+    console.log(error, 'err');
```
Comment (AdhamHaisami) [inline src/controllers/user/signup.ts]: use the custom error not just log the error

File: src/controllers/user/signup.ts [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
+import { hash } from 'bcrypt';
 import User from '../../database/user';
+// import hashPassword from '../../helpers/hashPromises';
 
 const signupController = async (req: Request, res: Response): Promise<void> => {
   try {
-    const { username, email, password, wieght, hieght, gender, goalWieght, role } = req.body;
+    const { username, email, password, weight, height, gender, goalWeight, role } = req.body;
+    const hashedPassword = await hash(password, 10);
     await User.create({
```
Comment (AdhamHaisami) [inline src/controllers/user/signup.ts]: where is the validation before creating a user into the db

Comment (AdhamHaisami) [reply src/controllers/user/signup.ts]: also you must check if the email exists in the database before adding new one, since the email is uniqe

**Review by AdhamHaisami — CHANGES_REQUESTED** [review]

File: src/controllers/user/signup.ts (hunk 4 above)
Comment (AhmedMughessib) [reply src/controllers/user/signup.ts]: i checked if the email exist, didn't do the validation yet

Comment (AhmedMughessib) [reply src/controllers/user/signup.ts]: the validation will be done by Abeer

**Review by HamdyPs — APPROVED** [review]

**Review by heleena-fadi — APPROVED** [review]

**Review by abeerAyad — APPROVED** [review]

**Review by heleena-fadi — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by AdhamHaisami — APPROVED** [review]

**Pull request merged**